### PR TITLE
Clean up python2 style code

### LINKED ISSUE #6890: Deprecate Python2
**Is your feature request related to a problem? Please describe.**
As January 1st approaches, support for "Python 2" gets to its end. By deprecating it we can free some machine/time on CI and maybe simplify our "dependency matrix"

Note that this issue is not meant to be executed before the January 1st 2020.

**Describe the solution you'd like**
- [ ] Remove py2 from the build matrix in Jenkins
- [ ] Remove Python 2 from setup.py
- [ ] Remove `six` from requirements.txt

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by alexrecuenco**
This branch works [on top of the PR by @venthur](https://github.com/docker/compose/pull/7031) rebasing the branch on top of `1.26.x`

- It removes support for python2, and early versions of python3 (<3.5). I am not sure if I left some requirements somewhere, I would appreciate if someone can make sure that is the case.
- As recommended by @graingert, I added `pyupgrade` to the pre-commit hooks to allow an easier transition for developers used to python2.

~NOT~ NOTE: The only change to code-style when transferring to python3 I am not sure if people would agree is the change from `lambda c: c.name` to the more modern use of the operator module with `attrgetter('name')`

Closes: #6890
Resolves #6890 

Signed-off-by: alexrecuenco <alejandrogonzalezrecuenco@gmail.com>

### CODE REVIEW

**Comment by alexrecuenco** [comment]
Note that, as mentioned by @ulyssessouza [(here)](https://github.com/docker/compose/pull/7031#issuecomment-589569087), we won't be merging on master, 

> to sync that with the next Ubuntu release

That is why I thought it would be appropriate to merge on top of 1.26.x, since this is a 1.26.x Goal anyway.

**Comment by ulyssessouza** [comment]
@alexrecuenco Could you please rebase so that we can include it in the next RC?

**Comment by alexrecuenco** [comment]
Do you mean to rebase on master? Or on `1.26.x`

**Comment by ulyssessouza** [comment]
@alexrecuenco Rebase with from `master`, pls

**Comment by alexrecuenco** [comment]
I don't think I can rename the branch, for that I would need to make a new PR, let me know if you prefer that. 

Other than that, it should be rebased now.

**Comment by ulyssessouza** [comment]
The renaming was just for the title of the PR. The branch name has no impact

File: setup.py [hunk 1]
```diff
@@ -50,7 +49,6 @@ def find_version(*file_paths):
     tests_require.append('mock >= 1.0.1, < 4')
 
 extras_require = {
-    ':python_version < "3.2"': ['subprocess32 >= 3.5.4, < 4'],
     ':python_version < "3.4"': ['enum34 >= 1.0.4, < 2'],
```
Comment (ulyssessouza) [inline setup.py:52]: This line should be also removed, since we do not support `python <= 3.3`

File: setup.py [hunk 2]
```diff
@@ -94,7 +92,7 @@ def find_version(*file_paths):
     install_requires=install_requires,
     extras_require=extras_require,
     tests_require=tests_require,
-    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*',
+    python_requires='>=3.0, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*',
```
Comment (ulyssessouza) [inline setup.py:95]:
```suggestion
    python_requires='>=3.4',
```

**Comment by alexrecuenco** [comment]
Thank you, I added your suggestion and rebased again on `docker/master`.

**Pull request merged**